addons/hw_posbox_homepage/static/src/app/Homepage.js

```diff
@@ -0,0 +1,113 @@
+/* global owl */
+
+import { SingleData } from "./components/SingleData.js";
+import { IotDevices } from "./components/IotDevices.js";
+import { FooterButtons } from "./components/FooterButtons.js";
+import { ServerDialog } from "./components/dialog/ServerDialog.js";
+import { WifiDialog } from "./components/dialog/WifiDialog.js";
+import useStore from "./hooks/useStore.js";
+import { UpdateDialog } from "./components/dialog/UpdateDialog.js";
+
+const { Component, xml, useState } = owl;
+
+export class Homepage extends Component {
+    static props = {};
+    static components = {
+        SingleData,
+        IotDevices,
+        FooterButtons,
+        ServerDialog,
+        WifiDialog,
+        UpdateDialog,
+    };
+
+    setup() {
+        this.store = useStore();
+        this.state = useState({ loading: true, lastUpdate: 0 });
+        this.data = useState({});
+
+        this.loadInitailData();
+        this.store.advanced = localStorage.getItem("showAdvanced") === "true";
+
+        setInterval(() => {
+            this.loadInitailData();
+        }, 10000);
+
+        setInterval(() => {
+            this.state.lastUpdate = Math.round((new Date().getTime() - this.store.update) / 1000);
+        }, 500);
+    }
+
+    async loadInitailData() {
+        try {
+            const data = await this.store.rpc({
+                url: "/hw_posbox_homepage/data",
+            });
+            this.data = data;
+            this.store.base = data;
+            this.state.loading = false;
+            this.store.update = new Date().getTime();
+        } catch {
+            console.warn("Error while fetching data");
+        }
+    }
+
+    toggleAdvanced() {
+        this.store.advanced = !this.store.advanced;
+        localStorage.setItem("showAdvanced", this.store.advanced);
+    }
+
+    static template = xml`
```

Comment:
is there not some way to have template XML in dedicated files ? :thinking:

Reply:
If you find it, I will be happy to know

Reply:
I do guess we can't use `web.assets_common` as the IoT does not initiate the JS backend as intended ?

Reply:
Cannot use assets, its a standalone owl application. We don't have any db on the IOT so no assets.

Reply:
correct me if I am wrong, but I thought there was 2 types of assets, the "views" that indeed require a database and the "static" ones which are loaded dynamically. They likely also require a database, but did you double check ?

Reply:
I'm currently thinking about that too, I don't really know, I will try